src/modlist.cpp

```diff
@@ -962,6 +962,40 @@ bool ModList::setActive(const QString &name, bool active)
   }
 }
 
+int ModList::setActive(const QStringList& names, bool active) {
+
+  // We only add indices for mods that exist (modIndex != UINT_MAX)
+  // and that can be enabled / disabled.
+  QList<unsigned int> indices;
+  for (const auto& name : names) {
+    auto modIndex = ModInfo::getIndex(name);
+    if (modIndex != UINT_MAX) {
+
+      // This check is not done by the bulk Profile::setModsEnabled, so we
+      // do it here.
+      ModInfo::Ptr modInfo = ModInfo::getByIndex(modIndex);
+      if (!modInfo->alwaysEnabled()) {
+        indices.append(modIndex);
```

Comment:
But both checks are already done in Profile::setModsEnabled() https://github.com/ModOrganizer2/modorganizer/blob/master/src/profile.cpp#L574
I would say that logging the error in case a passed mod isn't found like what the profile does is good since the plugin author could use the feedback that they are passing wrong stuff.

Reply:
The check on the size is done in `setModsEnabled`, but that was already the case with `setActive(QString)` and `setModEnabled()`, so I wanted to be consistent.

I completely overlook https://github.com/ModOrganizer2/modorganizer/blob/master/src/profile.cpp#L588, so that's my fault...

I thought about adding debug log, so I can do it. I still think it's a good idea to return the number of enabled/disabled mods since it can be used by the plugin.

Reply:
Ah, good point about the size, yeah for the sake of having a more accurate return value it's good to make the index check here.